launch/launch/actions/execute_process.py

```diff
@@ -382,7 +384,21 @@ def _shutdown_process(self, context, *, send_sigint):
         if self.__shutdown_received:
             # Do not handle shutdown more than once.
             return None
+
+        if self.__running and (self.process_details is None or self._subprocess_transport is None):
+            # Defer shut down if the process is not ready
+            context.register_event_handler(
+                OnProcessStart(
+                    on_start=lambda event, context:
+                    self._shutdown_process(context, send_sigint=send_sigint)))
+            return None
```

Comment:
@Blast545 I'd think this should happen once we're past checking for completion (line 409) and iff it's not `running` already (assuming `running` means precisely that without further considerations).

Reply:
If that block happens after `self.__shutdown_received = True`, then the deferring of the shutdown does not occur because of the first `if self.__shutdown_received:`. I didn't find any other way to arrange it.

Reply:
`shutdown_received` is set if already started or finished, you mean setting that variable in every one of those scenarios?

Reply:
Yes. As it stands, if you'd got a `_shutdown_process()` on a finished action, you'd try to kill it.

Reply:
With the current implementation if an action finished normally then the `shutdown_received` variable is set and the `_shutdown_process` does nothing. I think I'm not following your idea.